Can only create one metadata/line list template

## Describe the bug

It looks like I can only create one metadata/line list template. The first template creation is fine:

![image](https://user-images.githubusercontent.com/200517/56834055-043add00-6836-11e9-9d7e-e55a1e0ccb29.png)

But, when I next go to create a new template, pressing **Save**:

![image](https://user-images.githubusercontent.com/200517/56834154-3e0be380-6836-11e9-9678-66cf089ecc3d.png)

Nothing gets saved. The following exception gets thrown in Java:

```
26 Apr 2019 15:15:58,197 ERROR ca.corefacility.bioinformatics.irida.ria.web.errors.ExceptionHandlerController:117 - Could not read document: Can not construct instance of java.lang.Boolean from String value 'agDateColumnFilter': only "true" or "false" recognized
 at [Source: java.io.PushbackInputStream@40aa82fa; line: 1, column: 212] (through reference chain: ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate["fields"]->java.util.ArrayList[0]->ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.A
gGridColumn["filter"]); nested exception is com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not construct instance of java.lang.Boolean from String value 'agDateColumnFilter': only "true" or "false" recognized
 at [Source: java.io.PushbackInputStream@40aa82fa; line: 1, column: 212] (through reference chain: ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate["fields"]->java.util.ArrayList[0]->ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.A
gGridColumn["filter"])                                                                                                                   
org.springframework.http.converter.HttpMessageNotReadableException: Could not read document: Can not construct instance of java.lang.Boolean from String value 'agDateColumnFilter': only "true" or "false" recognized
 at [Source: java.io.PushbackInputStream@40aa82fa; line: 1, column: 212] (through reference chain: ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate["fields"]->java.util.ArrayList[0]->ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.A
gGridColumn["filter"]); nested exception is com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not construct instance of java.lang.Boolean from String value 'agDateColumnFilter': only "true" or "false" recognized
 at [Source: java.io.PushbackInputStream@40aa82fa; line: 1, column: 212] (through reference chain: ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate["fields"]->java.util.ArrayList[0]->ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.A
gGridColumn["filter"])                                                                                             
        at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.readJavaType(AbstractJackson2HttpMessageConverter.java:224)
        at org.springframework.http.converter.json.AbstractJackson2HttpMessageConverter.read(AbstractJackson2HttpMessageConverter.java:208)
        at org.springframework.web.servlet.mvc.method.annotation.AbstractMessageConverterMethodArgumentResolver.readWithMessageConverters(AbstractMessageConverterMethodArgumentResolver.java:197)
        at org.springframework.web.servlet.mvc.method.annotation.RequestResponseBodyMethodProcessor.readWithMessageConverters(RequestResponseBodyMethodProcessor.java:147)
        at org.springframework.web.servlet.mvc.method.annotation.RequestResponseBodyMethodProcessor.resolveArgument(RequestResponseBodyMethodProcessor.java:125)
        at org.springframework.web.method.support.HandlerMethodArgumentResolverComposite.resolveArgument(HandlerMethodArgumentResolverComposite.java:78)
        at org.springframework.web.method.support.InvocableHandlerMethod.getMethodArgumentValues(InvocableHandlerMethod.java:162)
...
Caused by: com.fasterxml.jackson.databind.exc.InvalidFormatException: Can not construct instance of java.lang.Boolean from String value 'agDateColumnFilter': only "true" or "false" recognized
 at [Source: java.io.PushbackInputStream@40aa82fa; line: 1, column: 212] (through reference chain: ca.corefacility.bioinformatics.irida.ria.web.linelist.dto.UIMetadataTemplate["fields"]->java.util.ArrayList[0]->ca.corefacility.bioinformatics.irida.ria.web.components.agGrid.A
gGridColumn["filter"])                                                                                                                     
        at com.fasterxml.jackson.databind.exc.InvalidFormatException.from(InvalidFormatException.java:55)                                                                                         
        at com.fasterxml.jackson.databind.DeserializationContext.weirdStringException(DeserializationContext.java:883)                                                    
        at com.fasterxml.jackson.databind.deser.std.StdDeserializer._parseBoolean(StdDeserializer.java:200)
```

## Steps to reproduce the problem
1. In **Line List** page, create a first initial metadata template.
2. Try to create a new template with different columns.
3. The error will occur here and no template will be saved.

## Expected behaviour

I expected to be able to save the new template.

## Additional context

Tested in **Chromium** and **Firefox** in Linux.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -38,6 +38,7 @@ Changes
 * [UI]: Fixes issue where importing an excel file that contained "Created Date" and "Modified Date" created metadata fields with those labels.
 * [REST]: Updated http status code returned (400 Bad Request) when uploading files to the wrong type of run
 * [UI/Developer]: Updated to jquery v3.4.0 to fix security issue.
+* [UI]: Fixed bug causing issues with saving Line List templates.
 
 0.22.0 to 19.01
 ----------------

--- a/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/components/agGrid/AgGridColumn.java
+++ b/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/components/agGrid/AgGridColumn.java
@@ -166,7 +166,7 @@ public class AgGridColumn {
 		this.headerCheckboxSelection = headerCheckboxSelection;
 	}
 
-	public String isFilter() {
+	public String getFilter() {
 		return filter;
 	}
 

--- a/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/components/agGrid/AgGridColumn.java
+++ b/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/components/agGrid/AgGridColumn.java
@@ -58,9 +58,9 @@ public class AgGridColumn {
 	private boolean headerCheckboxSelection;
 
 	/**
-	 * Suppress the filter on this column
+	 * Type of column filter.
 	 */
-	private boolean filter;
+	private String filter;
 
 	/**
 	 * Suppress the ability to resize this column
@@ -166,11 +166,11 @@ public class AgGridColumn {
 		this.headerCheckboxSelection = headerCheckboxSelection;
 	}
 
-	public boolean isFilter() {
+	public String isFilter() {
 		return filter;
 	}
 
-	public void setFilter(boolean filter) {
+	public void setFilter(String filter) {
 		this.filter = filter;
 	}
 

--- a/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java
+++ b/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListController.java
@@ -351,7 +351,6 @@ public class LineListController {
 		iconField.setLockPosition(true);
 		iconField.setCheckboxSelection(true);
 		iconField.setHeaderCheckboxSelection(true);
-		iconField.setFilter(false);
 		iconField.setResizable(false);
 		fields.add(0, iconField);
 

--- a/src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js
+++ b/src/main/webapp/resources/js/pages/projects/linelist/reducers/fields.js
@@ -44,6 +44,7 @@ function getColumnDefinition(col) {
 
   if (field === FIELDS.icons) {
     Object.assign(col, {
+      filter: undefined,
       cellRenderer: "IconCellRenderer"
     });
   } else if (type === TYPES.date) {

--- a/src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectLineListPageIT.java
+++ b/src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/projects/ProjectLineListPageIT.java
@@ -1,15 +1,16 @@
 package ca.corefacility.bioinformatics.irida.ria.integration.projects;
 
-import static org.junit.Assert.*;
-
 import org.junit.Test;
 import org.openqa.selenium.Dimension;
-
-import com.github.springtestdbunit.annotation.DatabaseSetup;
 
 import ca.corefacility.bioinformatics.irida.ria.integration.AbstractIridaUIITChromeDriver;
 import ca.corefacility.bioinformatics.irida.ria.integration.pages.LoginPage;
 import ca.corefacility.bioinformatics.irida.ria.integration.pages.projects.ProjectLineListPage;
+
+import com.github.springtestdbunit.annotation.DatabaseSetup;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
 
 /**
  * <p>
@@ -82,6 +83,19 @@ public class ProjectLineListPageIT extends AbstractIridaUIITChromeDriver {
 		// Switch back to new template
 		page.selectTemplate(TEMPLATE_NAME);
 		assertEquals("Should have 3 columns visible", 3, page.getNumberOfTableColumnsVisible());
+
+		// Test creating a second template
+		page.toggleMetadataField(5);
+		assertEquals("Should have 4 columns visible", 4, page.getNumberOfTableColumnsVisible());
+		page.saveMetadataTemplate("ANOTHER TESTING TEMPLATE");
+
+		// Switch back to new template
+		page.selectTemplate(TEMPLATE_NAME);
+		assertEquals("Should have 3 columns visible", 3, page.getNumberOfTableColumnsVisible());
+
+		// Switch back to new template
+		page.selectTemplate("ANOTHER TESTING TEMPLATE");
+		assertEquals("Should have 4 columns visible", 4, page.getNumberOfTableColumnsVisible());
 
 		// Test inline editing
 		page.selectTemplate("All Fields");

